src/storage/src/source/generator/counter.rs

```diff
@@ -13,7 +13,12 @@ use mz_ore::now::NowFn;
 use mz_repr::{Datum, Row};
 use mz_storage_client::types::sources::{Generator, GeneratorMessageType};
 
-pub struct Counter {}
+pub struct Counter {
+    /// How many values will be emitted
+    /// before old ones are retracted, or `None` for
+    /// an append-only collection.
+    pub max_cardinality: Option<usize>,
```

Comment:
Could this easily be a u64 instead? That'd match the proto stuff. Or does it make the math below harder?

Reply:
I made this a `usize` because it felt more semantically correct to me. It has to be converted to `u64` on the boundaries because protobuf doesn't have the concept of `usize`. I could also make it `u64`; I don't feel strongly.

Reply:
(We discussed that on Slack and it seems like it results in the simplest code)